Add Prcurve support (#12)
mlpack/mlboard: examples/prcurve.md

> ## Logging PrCurve

Reviewer: ```suggestion
## Logging PR-Curve
```

These examples help you to understand `SummaryWriter::Prcurve()` API in depth.

### 0. API 

  1. [Log PrCurve](#1-prcurve)
  2. [Log Prcurve stored in arma::vec](#2-prcurve-arma-vec)

> ### 1. PrCurve

Reviewer: ```suggestion
### 1. PR-Curve
```

> A embedding could be logged using the following API:

Reviewer: I think we are not logging embedding over here right?

```cpp
void PrCurve(const std::string tag,
             const std::vector<double>& labels,
             const std::vector<double>& predictions,

>              Filewriter& fw,

Reviewer: ```suggestion
             mlboard::Filewriter& fw,
```

             int threshold,
             std::vector<double>weights,
             const std::string& displayName,
             const std::string& description)
```

The API accepts `tag`, `labels`, `predictions`, `mlboard::Filewriter`, `thresholds`, `weights`, `displayName` and `description`.

Following is a snippet that would log some embedding values.

```cpp
#include <mlboard/mlboard.hpp>
#include <iostream>
#include <chrono> 
#include <ctime> 
#include <future>
int main()
{
  // Creating a FileWriter object that is responsible for logging the summary.
  std::chrono::time_point<std::chrono::system_clock> start, end; 
  start = std::chrono::system_clock::now(); 
  mlboard::FileWriter f1("temp");

>   // Log prcurve

Reviewer: ```suggestion
  // Log PR-Curve.
```

>   std::vector<double> labels = {1 ,1, 1, 1, 1 ,1 ,1 ,1 ,0 ,1};

Reviewer: ```suggestion
  std::vector<double> labels = {1, 1, 1, 1, 1, 1, 1, 1, 0, 1};
```

>   std::vector<double> predictions = {0.6458941 , 0.3843817, 0.4375872,

Reviewer: ```suggestion
  std::vector<double> predictions = {0.6458941, 0.3843817, 0.4375872,
```

      0.2975346, 0.891773, 0.05671298, 0.96366274, 0.2726563,

>       0.3834415,0.47766513};

Reviewer: ```suggestion
      0.3834415, 0.47766513};
```

  mlboard::SummaryWriter<mlboard::FileWriter>::PrCurve("test_pr_curve",

>       labels, predictions, *f1);

Reviewer: ```suggestion
      labels, predictions, f1);
```

Author: That was a very nice catch :)

  
  // This will allow you to indicate that you have logged all your data.
  f1.Close();
  end = std::chrono::system_clock::now(); 
  std::chrono::duration<double> elapsed_seconds = end - start; 
  std::time_t end_time = std::chrono::system_clock::to_time_t(end); 
  std::cout << "finished computation at " << std::ctime(&end_time) 
            << "elapsed time: " << elapsed_seconds.count() << "s\n"; 
}
```

The output would be similar to:

<p>
<img src = "assets/prcurve.jpg" width = "800" height = "400"/>
</p>

> ### 2. PrCurve Arma vec

Reviewer: ```suggestion
### 2. PR-Curve Arma vec
```

> You could log prcurve values stored in `arma::vec` using the following api:

Reviewer: ```suggestion
You could log prcurve values stored in `arma::vec` using the following API:
```

```cpp
void PrCurve(const std::string tag,
             const vecType& labels,
             const vecType& predictions,

>              Filewriter& fw,

Reviewer: ```suggestion
             mlboard::Filewriter& fw,
```

             int threshold,
             vecType weights,
             const std::string& displayName,
             const std::string& description)
```

Following is a snippet that would log some prcurve values stored in `arma::vec`.

```cpp
#include <mlboard/mlboard.hpp>
#include <iostream>
#include <chrono> 
#include <ctime> 
#include <future>
int main()
{
  // Creating a FileWriter object that is responsible for logging the summary.
  std::chrono::time_point<std::chrono::system_clock> start, end; 
  start = std::chrono::system_clock::now(); 
  mlboard::FileWriter f1("temp");
  // Log prcurve.

>   arma::rowvec labels = {1 ,1, 1, 1, 1 ,1 ,1 ,1 ,0 ,1};

Reviewer: ```suggestion
  arma::rowvec labels = {1, 1, 1, 1, 1, 1, 1, 1, 0, 1};
```

Reviewer: This is issue is still there.

  arma::rowvec predictions = {0.6458941 , 0.3843817, 0.4375872,
      0.2975346, 0.891773, 0.05671298, 0.96366274, 0.2726563,

>       0.3834415,0.47766513};

Reviewer: ```suggestion
      0.3834415, 0.47766513};
```

Reviewer: Same about this

  mlboard::SummaryWriter<mlboard::FileWriter>::PrCurve("test_pr_curve_arma_vec",

>       labels, predictions, *f1);

Reviewer: ```suggestion
      labels, predictions, f1);
```

  // This will allow you to indicate that you have logged all your data.
  f1.Close();
  end = std::chrono::system_clock::now(); 
  std::chrono::duration<double> elapsed_seconds = end - start; 
  std::time_t end_time = std::chrono::system_clock::to_time_t(end); 
  std::cout << "finished computation at " << std::ctime(&end_time) 
            << "elapsed time: " << elapsed_seconds.count() << "s\n"; 

> }

Reviewer: I am not sure. But I think you missed to add image at the end.

Author: It is the same image

## Logging PR-Curve

> These examples help you to understand `SummaryWriter::Prcurve()` API in depth.

Reviewer: ```suggestion
These examples help you to understand `SummaryWriter::PRcurve()` API in depth.
```

> These examples help you to understand `SummaryWriter::PRcurve()` API in depth.

Reviewer: ```suggestion
These examples help you to understand `SummaryWriter::PRCurve()` API in depth.
```

### 0. API 

>   1. [Log PrCurve](#1-pr-curve)

Reviewer: ```suggestion
  1. [Log PR-Curve](#1-pr-curve)
```

>   2. [Log Prcurve stored in arma::vec](#2-pr-curve-arma-vec)

Reviewer: ```suggestion
  2. [Log PR-Curve stored in arma::vec](#2-pr-curve-arma-vec)
```

### 1. PR-Curve

> A PR Curve could be logged using the following API:

Reviewer: ```suggestion
A PR-Curve could be logged using the following API:
```

```cpp
template<typename Filewriter>

> void SummaryWriter<Filewriter>::PrCurve(

Reviewer: ```suggestion
void SummaryWriter<Filewriter>::PRCurve(
```

    const std::string tag,
    const std::vector<double>& labels,
    const std::vector<double>& predictions,
    mlboard::Filewriter& fw,
    int threshold,
    std::vector<double>weights,
    const std::string& displayName,
    const std::string& description)
```

The API accepts `tag`, `labels`, `predictions`, `mlboard::Filewriter`, `thresholds`, `weights`, `displayName` and `description`.

> Following is a snippet that would log some PR curve.

Reviewer: ```suggestion
Following is a snippet that would log some PR-Curve.
```

> Following is a snippet that would log some embedding values.

Reviewer: I think over here you missed to replace the embedding line to PR-Curve specific.

```cpp
#include <mlboard/mlboard.hpp>
#include <iostream>
#include <chrono> 
#include <ctime> 
#include <future>
int main()
{
  // Creating a FileWriter object that is responsible for logging the summary.
  std::chrono::time_point<std::chrono::system_clock> start, end; 
  start = std::chrono::system_clock::now(); 
  mlboard::FileWriter f1("temp");
  // Log PR-Curve.
  std::vector<double> labels = {1, 1, 1, 1, 1, 1, 1, 1, 0, 1};
  std::vector<double> predictions = {0.6458941, 0.3843817, 0.4375872,
      0.2975346, 0.891773, 0.05671298, 0.96366274, 0.2726563,
      0.3834415, 0.47766513};

>   mlboard::SummaryWriter<mlboard::FileWriter>::PrCurve("test_pr_curve",

Reviewer: ```suggestion
  mlboard::SummaryWriter<mlboard::FileWriter>::PRCurve("test_pr_curve",
```

      labels, predictions, f1);
  
  // This will allow you to indicate that you have logged all your data.
  f1.Close();
  end = std::chrono::system_clock::now(); 
  std::chrono::duration<double> elapsed_seconds = end - start; 
  std::time_t end_time = std::chrono::system_clock::to_time_t(end); 
  std::cout << "finished computation at " << std::ctime(&end_time) 
            << "elapsed time: " << elapsed_seconds.count() << "s\n"; 
}
```

The output would be similar to:

<p>
<img src = "assets/prcurve.jpg" width = "800" height = "400"/>
</p>

### 2. PR-Curve Arma vec

> You could log prcurve values stored in `arma::vec` using the following API:

Reviewer: ```suggestion
You could log PR-Curve values stored in `arma::vec` using the following API:
```

```cpp

> void PrCurve(const std::string tag,

Reviewer: ```suggestion
void PRCurve(const std::string tag,
```

             const vecType& labels,
             const vecType& predictions,
             mlboard::Filewriter& fw,
             int threshold,
             vecType weights,
             const std::string& displayName,
             const std::string& description)
```

> Following is a snippet that would log some prcurve values stored in `arma::vec`.

Reviewer: ```suggestion
Following is a snippet that would log some PR-Curve values stored in `arma::vec`.
```

```cpp
#include <mlboard/mlboard.hpp>
#include <iostream>
#include <chrono> 
#include <ctime> 
#include <future>
int main()
{
  // Creating a FileWriter object that is responsible for logging the summary.
  std::chrono::time_point<std::chrono::system_clock> start, end; 
  start = std::chrono::system_clock::now(); 
  mlboard::FileWriter f1("temp");
  // Log prcurve.
  arma::rowvec labels = {1, 1, 1, 1, 1, 1, 1, 1, 0, 1};
  arma::rowvec predictions = {0.6458941 , 0.3843817, 0.4375872,
      0.2975346, 0.891773, 0.05671298, 0.96366274, 0.2726563,
      0.3834415, 0.47766513};

>   mlboard::SummaryWriter<mlboard::FileWriter>::PrCurve("test_pr_curve_arma_vec",

Reviewer: ```suggestion
  mlboard::SummaryWriter<mlboard::FileWriter>::PRCurve("test_pr_curve_arma_vec",
```
